### PR TITLE
Quick fix

### PR DESCRIPTION
**Pull request by AliabbasMerchant**
@alexander255 These are the fixes as discussed under #1   
No need to review them separately.  
Please merge #2 before merging this.

### CODE REVIEW

File: datastore/core/key.py [hunk 1]
```diff
@@ -190,12 +190,12 @@ def __len__(self):
     def __lt__(self, other):
         if isinstance(other, Key):
             return self._string < other._string
-        raise TypeError('other is not of type %s' % Key)
+        raise NotImplementedError('The second operand is not of type %s' % Key)
```
Comment (ntninja) [inline datastore/core/key.py]: This should `return NotImplemented` (a special built-in type), not directly raise `NotImplementedError`. The later being automatically raised by the Python interpreter when all possible combinations of the comparison (`A.__lt__(B)` and `B.__ge__(A)`) have been exhausted.

File: datastore/core/key.py [hunk 2]
```diff
@@ -190,12 +190,12 @@ def __len__(self):
...
 
     def __eq__(self, other):
         if isinstance(other, Key):
             return self._string == other._string
-        return False
+        raise NotImplementedError('The second operand is not of type %s' % Key)
```
Comment (ntninja) [inline datastore/core/key.py]: Same here.

File: datastore/core/query.py [hunk 3]
```diff
@@ -235,7 +237,7 @@ def __repr__(self):
         return "Order('%s%s')" % (self.op, self.field)
 
     def __eq__(self, other):
-        return self.field == other.field and self.op == other.op
+        return self.field == other.type and self.op == other.op
```
Comment (ntninja) [inline datastore/core/query.py]: Already mentioned this in other PR, please do point out if I'm wrong!

File: datastore/core/serialize.py [hunk 4]
```diff
@@ -10,17 +11,12 @@ class Serializer(object):
     @classmethod
     def loads(cls, value):
         """returns deserialized `value`."""
-        raise NotImplementedError
+        raise NotImplementedError()
```
Comment (ntninja) [inline datastore/core/serialize.py]: *Note*: Here it's fine to raise however, because `.loads` is not a special Python method like `__lt__`

**Review by ntninja — COMMENTED** [review]
Basically just the thing from the other PR and the nit about using `return NotImplemented`.

Thanks!

File: datastore/core/key.py (hunk 1 above)
Comment (AliabbasMerchant) [reply datastore/core/key.py]: Cool

File: datastore/core/serialize.py (hunk 4 above)
Comment (AliabbasMerchant) [reply datastore/core/serialize.py]: Okay. Got it 👍 Thanks

**Comment by AliabbasMerchant** [comment]
@alexander255 I have made those required changes

File: datastore/core/serialize.py [hunk 5]
```diff
@@ -1,22 +1,24 @@
...
 
 default_serializer = json
 
 
-class Serializer(object):
+class Serializer(ABC):
     """Serializing protocol. Serialized data must be a string."""
 
     @classmethod
+    @abstractmethod
```
Comment (ntninja) [inline datastore/core/serialize.py:12]: Nice! :slightly_smiling_face:

**Pull request merged**